test/unit/assorted/server_discovery_and_monitoring.spec.test.ts

```diff
@@ -361,25 +356,29 @@ function assertOutcomeExpectations(topology, events, outcome) {
         expect(actualEvent).to.matchMongoSpec(expectedEvent);
       }
 
-      return;
+      continue;
     }
 
     if (key === 'compatible' || key === 'setName') {
       if (outcomeValue == null) {
-        expect(topology.description[key]).to.not.exist;
+        expect(topology.description[key]).to.be.undefined;
       } else {
         expect(topology.description).property(key).to.equal(outcomeValue);
       }
 
-      return;
+      continue;
     }
 
-    expect(description).to.include.keys(translatedKey);
+    expect(description).to.include.keys(key);
 
     if (outcomeValue == null) {
-      expect(description[translatedKey]).to.not.exist;
+      expect(description[key]).to.be.undefined;
     } else {
-      expect(description).property(translatedKey).to.eql(outcomeValue, `(key="${translatedKey}")`);
+      if (typeof outcomeValue === 'object') {
+        expect(description).to.have.property(key).that.deep.equals(outcomeValue);
```

Comment:
you can use `deep.property(key, outcomeValue)` here for deep comparison

Reply:
Changed!